woocommerce/woocommerce
# Update minimal wp dependencies to fix some typing issues when building the monorepo

### Changes proposed in this Pull Request:

The past week I have been blocked every single time I wanted to update any dependency with the `@wordpress` prefix (in any monorepo package). I tried to get to the bottom of the issue and a recurring problem was with the `product-editor` package types.

`product-editor` would go back and forth between resolving DT types for wp core-data and newer ones (despite depending on the older DT types). This happened any time I updated any wp dependency in any package. So I have to resolve these issues as part of any first dependency i successfully update.

With this PR I am hoping to break the ice on updating the WP depdencies, unblock the product-editor related errors and at least update one package. So i tackled a single package update: `@wordpress/core-data` in the data package. I've updated it to the latest wp tag of 6.5. I'm hoping this will unblock me to change a couple more dependencies in other packages without `product-editor` immediately breaking.

I had already done some of the updates here as part of a large PR a while ago, but I realized I am going to need to do this in small pieces if I expect to get anything merged because otherwise there are just too many changes to feel confident about doing in a single PR.

#### Summary of what I did:

1. Update `@woocommerce/data` dependency on core/data to wp-6.5. Fix the type errors that I've already done before as a subset of this PR: https://github.com/woocommerce/woocommerce/pull/43494
2. Fix the subsequently raised product-editor type errors that I've seen 100 times now. I hope these will stop coming up now as we update more packages.
3. in wc-admin I was forced to add 2 ts-ignore's. I briefly explored updating any types here but it would have begun a cascading type hell again so I've avoided it
4. Add a temp syncpack rule to ignore pinning wp dependencies in wc/data. I think to slowly update all the packages Syncpack will need to eventually be disabled for wp types in the monorepo until we can get them updated everywhere. I would be happy to change this for just removing the wp rule from syncpack for now if preferred.






### How to test the changes in this Pull Request:

There are no runtime changes in this PR, the actual packages resolved on the editor are controlled by WP itself, so any package update here is just for the sake of more accurate typing. Also all the actual code changes are just TS type changes not runtime changes.

**To test this, I'd just like to see all the builds and CI pass.**



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 
Update minimal wp dependencies to fix TS type issues when building the monorepo

</details>
